app/recordtransfer/jobs.py

```diff
@@ -88,18 +89,40 @@ def create_downloadable_bag(submission: Submission, user_triggered: User):
 
 @django_rq.job
 def cleanup_expired_sessions() -> None:
-    """Delete all UploadSession objects that have expired."""
-    LOGGER.info("Deleting expired upload sessions ...")
+    """Clean up UploadSession objects that are expirable. Upload sessions that are not associated
+    with any InProgressSubmission objects are deleted, while those that are associated with
+    InProgressSubmission objects have their uploads removed and are expired.
+    """
+    LOGGER.info("Cleaning up upload sessions ...")
     try:
-        expired_sessions = UploadSession.objects.get_expired().all()
-
-        if expired_sessions.count() == 0:
-            LOGGER.info("No expired upload sessions to delete")
+        expirable_sessions: QuerySet[UploadSession] = UploadSession.objects.get_expirable().all()
+        deletable_sessions: QuerySet[UploadSession] = UploadSession.objects.get_deletable().all()
+        if expirable_sessions.count() == 0 and deletable_sessions.count() == 0:
+            LOGGER.info("No expired upload sessions to clean up")
             return
 
-        expired_sessions.delete()
+        deleted = 0
+        expired = 0
+        for session in expirable_sessions:
+            # Check if there are any InProgressSubmission objects associated with this session.
+            if InProgressSubmission.objects.filter(upload_session=session).exists():
+                session.remove_temp_uploads()
+                session.expire()
+                expired += 1
+            else:
+                # Session is in limbo, delete it.
+                session.delete()
+                deleted += 1
+        for session in deletable_sessions:
+            session.delete()
+            deleted += 1
```

Comment:
If `get_deletable` and `get_expirable` both return completely separate upload sessions, then we can simplify this part of the job. I'm also recommending we move `remove_temp_uploads()` inside `expire()`.

```suggestion
        deleted = deletable_sessions.count()
        expired = expirable_sessions.count()
 
        for session in expirable_sessions:
            session.expire()

        for session in deletable_sessions:
            session.delete()
```

Reply:
Addressed by: [Simplify session processing for cleanup](https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/431/commits/7b9f429218a5d3b3dfa44c59d6599834d401a7c3)